### PR TITLE
Tooltips of metrics don't work

### LINKED ISSUE #727: Tooltips of metrics don't work
When hovering over the question marks of the `Metrics`, it should show a tooltip with the `helpText` but it doesn't work anymore.

image

### PR DESCRIPTION
**Pull request by JonasKellerer**
Resolves #727

### CODE REVIEW

File: src/components/VariantMutationComparison/VariantMutationComparisonHook.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      mutationsFromVariant1,
+      mutationsFromVariant2
+    );
+    const genes = groupByGenes(sharedMutations, mutationsOnlyInVariant1, mutationsOnlyInVariant2);
+    // Count
+    return [...genes.entries()].map(([gene, mutations]) => ({
+      gene,
+      ...mutations,
+    }));
+  }, [selectedVariants.data, minProportion]);
```
Comment (fengelniederhammer) [inline src/components/VariantMutationComparison/VariantMutationComparisonHook.ts]: IntelliJ: "ESLint: React Hook useMemo has a missing dependency: 'selectedVariants'. Either include it or remove the dependency array.(react-hooks/exhaustive-deps)"

File: src/components/VariantMutationComparison/VariantMutationComparison.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+function VariantTableEntryWithTooltip(mutations: string[]) {
+  return mutations.length === 0 ? (
+    mutations.length
+  ) : (
+    <CustomWidthTooltip title={sortAAMutationList(mutations).join(', ')} followCursor>
+      <div>{mutations.length}</div>
+    </CustomWidthTooltip>
+  );
+}
```
Comment (fengelniederhammer) [inline src/components/VariantMutationComparison/VariantMutationComparison.tsx]:
```suggestion
function VariantTableEntryWithTooltip({ mutations }: { mutations: string[] }) {
  if (mutations.length === 0) {
    return <>{mutations.length}</>;
  }
  return (
    <CustomWidthTooltip title={sortAAMutationList(mutations).join(', ')} followCursor>
      <div>{mutations.length}</div>
    </CustomWidthTooltip>
  );
}
```

Then we could use it like a proper component.

File: src/components/VariantMutationComparison/__test__/VariantMutationComparison.test.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      mutationsOnlyIn1: [],
+      mutationsOnlyIn2: [],
+      sharedMutations: [],
+      gene: 'E',
+    };
+
+    renderVariantTable([mutationOnGene], ['SOMEVARIANT', 'OTHERVARIANT']);
+
+    expect(screen.getByText('50%')).toBeInTheDocument();
+    expect(screen.queryByRole('tooltip')).toBeNull();
```
Comment (fengelniederhammer) [inline src/components/VariantMutationComparison/__test__/VariantMutationComparison.test.tsx]:
```suggestion
    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
```

File: src/components/VariantMutationComparison/index.tsx [hunk 4]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (fengelniederhammer) [inline src/components/VariantMutationComparison/index.tsx]: Reexport `VariantMutationComparison` here

**Comment by fengelniederhammer** [comment]
Looks good to me

**Review by chaoran-chen — APPROVED** [review]
Thanks! I can confirm that the functionality is as expected. Since Fabian has reviewed the code, I'm happy to have it merged!

**Pull request merged**